app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -132,11 +134,12 @@ public function indexAction()
             // Check session cache for estimate.  If the shipping city or postcode, and the country code match,
             // then use the cached version.  Otherwise, we have to do another calculation
             ////////////////////////////////////////////////////////////////////////////////////////
-            $cached_address = unserialize(Mage::app()->getCache()->load("quote_location_".$quote->getId()));
+            $cacheKey = $this->generateCacheKey($quote->getId(), $quote->getShippingAddress(), $quote->getGrandTotal());
```

Comment:
should we also include if user is logged into magento? (I assume special customer class has different shipping treatment?)

Reply:
Also we probably need to consider discount - freeshipping discount don't change ground total but changes shipping